presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestTpchCostBasedPlan.java

```diff
@@ -0,0 +1,110 @@
+ */
+
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.Session.SessionBuilder;
+import com.facebook.presto.benchto.JustToAccessPrestoBenchtoQueriesResources;
+import com.facebook.presto.sql.analyzer.FeaturesConfig.JoinDistributionType;
+import com.facebook.presto.sql.analyzer.FeaturesConfig.JoinReorderingStrategy;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static com.facebook.presto.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchConnectorFactory.TPCH_COLUMN_NAMING_PROPERTY;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
+
+/**
+ * This class tests cost-based optimization rules related to joins. It contains unmodified TPCH queries.
+ * This class is using TPCH connector configured in way to mock Hive connector with unpartitioned TPCH tables.
+ */
+public class TestTpchCostBasedPlan
+        extends BaseCostBasedPlanTest
+{
+    /*
+     * CAUTION: The expected plans here are not necessarily optimal yet. Their role is to prevent
+     * inadvertent regressions. A conscious improvement to the planner may require changing some
+     * of the expected plans, but any such change should be verified on an actual cluster with
+     * large amount of data.
+     */
+
+    public TestTpchCostBasedPlan()
+    {
+        super(() -> {
+            String catalog = "local";
+            SessionBuilder sessionBuilder = testSessionBuilder()
+                    .setCatalog(catalog)
+                    .setSchema("sf3000.0")
+                    .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
+                    .setSystemProperty(JOIN_REORDERING_STRATEGY, JoinReorderingStrategy.AUTOMATIC.name())
+                    .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.AUTOMATIC.name());
+
+            LocalQueryRunner queryRunner = LocalQueryRunner.queryRunnerWithFakeNodeCountForStats(sessionBuilder.build(), 8);
+            queryRunner.createCatalog(
+                    catalog,
+                    new TpchConnectorFactory(1, false, false),
+                    ImmutableMap.of(TPCH_COLUMN_NAMING_PROPERTY, ColumnNaming.SIMPLIFIED.name()));
+            return queryRunner;
+        });
+    }
+
+    @Override
+    protected Stream<Query> getQueries()
+    {
+        return IntStream.rangeClosed(1, 22)
```

Comment:
Scan queries in the classpath instead. 

```
Reflections reflections = new Reflections("some.package", new ResourcesScanner());
Set<String> fileNames = reflections.getResources(Pattern.compile("q(\\d+)\.sql"));
```

Reply:
I would prefer not to drag new dependency just for this

Reply:
Sorry, i thought this is the class from Guava. Nevermind than.